### PR TITLE
Cross protocol shovel documentation.

### PR DESCRIPTION
**Pull request by kjnilsson**
Adding documentation for AMQP 1.0 and AMQP 0.9.1 endpoints.

[#138600475]

Part of: https://github.com/rabbitmq/rabbitmq-shovel/issues/26

### CODE REVIEW

File: site/shovel-dynamic.xml [hunk 1]
```diff
@@ -192,60 +270,54 @@ limitations under the License.
...
+      <dl>
+        <dt><code>src-uri</code></dt>
         <dd>
           <p>
-            Determines how the shovel should acknowledge messages. If set
-            to <code>on-confirm</code> (the default), messages are
-            acknowledged to the source broker after they have been confirmed
-            by the destination. This handles network errors and broker
-            failures without losing messages, and is the slowest option.
+            The AMQP URI(s) for the source. Mandatory.
```
Comment (dcorbacho) [inline site/shovel-dynamic.xml]: Should refer to `src-uri` in 0.9.1 source for details

File: site/shovel-dynamic.xml [hunk 2]
```diff
@@ -258,18 +330,55 @@ limitations under the License.
...
         </dd>
       </dl>
+      <h4>AMQP 1.0 Destination</h4>
+
+      <dl>
+        <dt><code>dest-uri</code></dt>
+        <dd>
+          <p>
+            The AMQP URI(s) for the
+            destination. Mandatory. See <code>src-uri</code> above. Mandatory.
```
Comment (dcorbacho) [inline site/shovel-dynamic.xml]: Pointing to `src-uri` above is confusing, as just above is a 1-line description in the AMQP 1.0 source. They both should refer to the long description in 0.9.1 source.

File: site/shovel-dynamic.xml [hunk 3]
```diff
@@ -192,60 +270,54 @@ limitations under the License.
...
-            Determines how the shovel should acknowledge messages. If set
-            to <code>on-confirm</code> (the default), messages are
-            acknowledged to the source broker after they have been confirmed
-            by the destination. This handles network errors and broker
-            failures without losing messages, and is the slowest option.
+            The AMQP URI(s) for the source. Mandatory.
           </p>
+        </dd>
+
+        <dt><code>src-address</code></dt>
```
Comment (dcorbacho) [inline site/shovel-dynamic.xml]: This is a c&p from `src-queue`

File: site/shovel-dynamic.xml (hunk 1 above)
Comment (kjnilsson) [inline site/shovel-dynamic.xml]: addressed

**Pull request merged**